lib/filewatch/settings.rb

```diff
@@ -51,7 +47,7 @@ def add_options(opts)
       @file_chunk_count = @opts[:file_chunk_count]
       @sincedb_path = @opts[:sincedb_path]
       @sincedb_write_interval = @opts[:sincedb_write_interval]
-      @sincedb_expiry_duration =  self.class.days_to_seconds(@opts.fetch(:sincedb_clean_after, 14))
+      @sincedb_expiry_duration =  @opts.fetch(:sincedb_clean_after)
```

Comment:
there are 2 different paths hitting this: the first one has `sincedb_clean_after => 14` thus fine
but the second (creating an `ObservingRead` or `ObservingTail`) isn't explicit about `:sincedb_clean_after`
... https://github.com/logstash-plugins/logstash-input-file/blob/v4.1.14/lib/filewatch/observing_base.rb#L55
```ruby
      options = {
        :sincedb_write_interval => 10,
        :stat_interval => 1,
        :discover_interval => 5,
        :exclude => [],
        :start_new_files_at => :end,
        :delimiter => "\n",
        :file_chunk_count => MAX_ITERATIONS,
        :file_chunk_size => FILE_READ_SIZE,
        :file_sort_by => "last_modified",
        :file_sort_direction => "asc",
      }.merge(opts)
      unless options.include?(:sincedb_path)
        raise NoSinceDBPathGiven.new("No sincedb_path set in options. This should have been added in the main LogStash::Inputs::File class")
      end
      @settings = Settings.from_options(options)
```
... haven't checked if that has good enough coverage by specs 
but sounds like an issue not having the `14` default, wdyt?

Reply:
I was thinking the same, but the looking at https://github.com/logstash-plugins/logstash-input-file/blob/v4.1.14/lib/filewatch/settings.rb#L13 you have 2 calls:
- initialize
- add_options

The initialize instantiate a default hash and executes `add_options` on that default populating the `@sincedb_expiry_duration` with the value of `days_to_seconds(14)` the second time, on the instance of `Settings` just created it calls `add_options` but this time on the opts provided externally and fetch the value from this map and store it in  `@sincedb_expiry_duration` overwriting the previous value https://github.com/logstash-plugins/logstash-input-file/blob/v4.1.14/lib/filewatch/settings.rb#L54.

This second value is the value of the plugin's configuration field https://github.com/logstash-plugins/logstash-input-file/blob/v4.1.14/lib/logstash/inputs/file.rb#L195 and the value to `FriendlyDuration` is alredy converted into seconds by https://github.com/logstash-plugins/logstash-input-file/blob/v4.1.14/lib/logstash/inputs/friendly_durations.rb#L17-L43

Reply:
thanks @andsel in that case I believe we're all covered here.